lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -8,13 +8,33 @@ const { allAuthedUsersId, bucketOwnerActions, logId, publicId,
 const publicReadBuckets = process.env.ALLOW_PUBLIC_READ_BUCKETS ?
     process.env.ALLOW_PUBLIC_READ_BUCKETS.split(',') : [];
 
-function checkBucketAcls(bucket, requestType, canonicalID) {
+function checkBucketAcls(bucket, requestType, canonicalID, mainApiCall) {
+    // Same logic applies on the Versioned APIs, so let's simplify it.
+    const requestTypeParsed = requestType.endsWith('Version') ?
+        requestType.slice(0, -7) : requestType;
     if (bucket.getOwner() === canonicalID) {
         return true;
     }
+    // Backward compatibility
+    const arrayOfAllowed = [
+        'objectGetTagging',
+        'objectPutTagging',
+        'objectPutLegalHold',
+        'objectPutRetention',
+    ];
+    if (mainApiCall === 'objectGet') {
+        if (arrayOfAllowed.includes(requestTypeParsed)) {
```

Comment:
if mainApiCall is objectGet, we should only accept requestTypeParsed == objectGetTagging, right?

(having the list for checking the different types of requests for objectPut is fine, this is really just about the "mixing" the Get and Put...)

Reply:
Yes seperating these makes sense, I'll change this.
92cbbdb44f32f738006ff3523641e2de7c65dd9d